File: mcorg.backends/common/src/main/java/app/mcorg/common/domain/model/Authority.java
package app.mcorg.common.domain.model;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum Authority {
    OWNER(10_000),
    ADMIN(5_000),
    PARTICIPANT(1_000);

    private final int rank;

    public boolean equalsOrHigher(Authority other) {
        return rank >= other.rank;
    }

    public boolean equalsOrLower(Authority other) {
        return rank <= other.rank;
    }
}

File: webapp/src/main/kotlin/app/mcorg/domain/model/permissions/Authority.kt
package app.mcorg.domain.model.permissions

enum class Authority(val level: Int) {
    OWNER(0),
    ADMIN(10),
    PARTICIPANT(20);

    companion object {
        fun fromLevel(level: Int): Authority? {
            return values().firstOrNull { it.level == level }
        }
    }
}